brave/brave-core
# Decouples credentials logic from promotions

Resolves https://github.com/brave/brave-browser/issues/8677

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
Plan 1:
- enable rewards
- claim promotion
- make sure that balance updates
- do a tip
- make sure that balance updates

Plan 2:
- make sure that VG claim to uphold works

Plan 3:
- enable rewards
- claim promotion
- close browser
- go into db and change `expires_at` of `promotion` table in the past, go to `unblinded_tokens` table and change `expires_at` to be in the past
- start browser
- make sure that balance is 0 and promotion expired

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
